src/test/abilities/stall.test.ts

```diff
@@ -0,0 +1,119 @@
+import { MovePhase } from "#app/phases";
+import GameManager from "#test/utils/gameManager";
+import { getMovePosition } from "#test/utils/gameManagerUtils";
+import { Abilities } from "#enums/abilities";
+import { Moves } from "#enums/moves";
+import { Species } from "#enums/species";
+import Phaser from "phaser";
+import { afterEach, beforeAll, beforeEach, describe, expect, it } from "vitest";
+
+
+describe("Ability - Stall", () => {
+  let phaserGame: Phaser.Game;
+  let game: GameManager;
+
+  beforeAll(() => {
+    phaserGame = new Phaser.Game({
+      type: Phaser.HEADLESS,
+    });
+  });
+
+  afterEach(() => {
+    game.phaseInterceptor.restoreOg();
+  });
+
+  beforeEach(() => {
+    game = new GameManager(phaserGame);
+    game.override.battleType("single");
+    game.override.disableCrits();
+    game.override.enemySpecies(Species.REGIELEKI);
+    game.override.enemyAbility(Abilities.STALL);
+    game.override.enemyMoveset([Moves.QUICK_ATTACK, Moves.QUICK_ATTACK, Moves.QUICK_ATTACK, Moves.QUICK_ATTACK]);
+    game.override.moveset([Moves.QUICK_ATTACK, Moves.TACKLE]);
+  });
+
+  /**
+   * Bulbapedia References:
+   * https://bulbapedia.bulbagarden.net/wiki/Stall_(Ability)
+   * https://bulbapedia.bulbagarden.net/wiki/Priority
+   **/
+
+  it("Pokemon with Stall should move last in its priority bracket regardless of speed", async() => {
+    await game.startBattle([ Species.SHUCKLE ]);
+
+    const leadPokemon = game.scene.getPlayerPokemon()!;
+    const enemyPokemon = game.scene.getEnemyPokemon()!;
+
+    const leadIndex = leadPokemon.getBattlerIndex();
+    const enemyIndex = enemyPokemon.getBattlerIndex();
```

Comment:
```suggestion
    const leadIndex = game.scene.getPlayerPokemon()!.getBattlerIndex();
    const enemyIndex = game.scene.getEnemyPokemon()!.getBattlerIndex();
```
NIT: Because we don't care much about the actual Pokemon themselves, you can just have variables for the indices alone (2).